### ISSUE TITLE
提示错误

### ISSUE THREAD

**Issue by lililii124** [CMU-CBML/HybridOctree_Hex#3]
![Quicker_20240618_204103](https://github.com/CMU-CBML/HybridOctree_Hex/assets/64952090/a5c7d6d6-45c9-4e11-a12e-faf8df436f07)
![Quicker_20240618_204103](https://github.com/CMU-CBML/HybridOctree_Hex/assets/64952090/74b95c2b-1618-4eae-bc07-ee6299497b62)

**Comment by gt2001** [comment]
I am not very sure why this error occurs. Are you running on a demo model I provided?

**Comment by yuyuanzi-hub** [comment]
Hello, thanks for author's work and your question. I just met the same problem, and have you solved it?

**Comment by yuyuanzi-hub** [comment]
![image](https://github.com/user-attachments/assets/a9d704c0-de69-403c-b249-171a65d7f67f)
within the first loop, z is 3861 and zz is 61776, maybe this part causes the vector out of range

**Comment by gt2001** [comment]
Thanks for your report. I recently received several reports that the code cannot finish normally. It seems that v1.3 has some problems. Can you try v1.2?

**Comment by yuyuanzi-hub** [comment]
Thank you for answering, will try

**Comment by yuyuanzi-hub** [comment]
Hello, I've tried v1.2 yesterday. It didn't create the the file of **DualHexFileName**, and the result was not complete. Maybe in this screenshot, these 2 file should be the same one? 
![image](https://github.com/user-attachments/assets/b501f204-5856-495d-acee-0043b28c581c)
![image](https://github.com/user-attachments/assets/6b360bea-ac20-41ec-96e5-68210e0b8d08)

**Comment by gt2001** [comment]
Can you set progress=0 and run again? If progress > 0, and the checkpoint file is missing, it will report the file missing error.

**Comment by yuyuanzi-hub** [comment]
Yes, I just set it 0, and ran it again. But still, it didn't create a dualhex file. 
Besides, I want ask the code won't stop itself right? It gave lines of badElem.
Thank you for your patience.

**Comment by gt2001** [comment]
For the second question, it won't stop itself, you can just take the result file when you think it is enough good.
For the first question, I tested the code again, it should be working normally. Can you check the function "RemoveOutsideElement" to see if it is working (especially the last line octreeMesh.WriteToVtk(fileName, BOX_LENGTH_RATIO, START_POINT);)? If this works, it should output dualHex.vtk

**Comment by yuyuanzi-hub** [comment]
I got you. Will check, thank you.

**Comment by yuyuanzi-hub** [comment]
It worked well. I'm looking forward to the faster version and hoping you are doing well.
I also want to know how to generate the result like your paper, with inside cube?

**Comment by gt2001** [comment]
I am glad you can replicate the result. The coming version focus on quality guarantee that ensures all positive Jacobian without optimization. The result is visualized with paraview box cutting and assigning different colors to inside and outside faces.

**Comment by yuyuanzi-hub** [comment]
Thank you for the settings in paraview.
